packages/synapse-react-client/src/components/HasAccess/HasAccessV2.tsx

```diff
@@ -18,18 +18,22 @@ import AccessRequirementList, {
   checkHasUnsupportedRequirement,
 } from '../AccessRequirementList/AccessRequirementList'
 import IconSvg, { IconName } from '../IconSvg/IconSvg'
-import { Theme, useTheme } from '@mui/material'
+import { Box, Button, Theme, useTheme } from '@mui/material'
 import { isFileEntity } from '../../utils/types/IsType'
 
 export type HasAccessProps = {
   onHide?: () => void
   entityId: string
   entityVersionNumber?: string
   className?: string
+  showButtonText?: boolean
 }
 
+const buttonSx = { p: '0px', minWidth: 'unset' }
+
 export enum FileHandleDownloadTypeEnum {
   Accessible = 'Accessible',
+  AccessibleWithTerms = 'AccessibleWithTerms',
```

Comment:
Ah, because the only difference before was the presence of the `View Terms` text, but now that that's gone, we need to add a completely new state. Makes sense!